File: python/src/adapter/driving/user_manage_controller.py
import logging

from fastapi import APIRouter, Request

from src.adapter.driving import request_auth_helper, request_value_helper
from src.adapter.driving.result import Result
from src.application.user_manage_service import UserManageService
from src.application.user_query_criteria import UserQueryCriteria
from src.application.user_query_handler import UserQueryHandler
from src.ioc_container import ioc_container

logger = logging.getLogger(__name__)
router = APIRouter()

user_manage_service = ioc_container.resolve(UserManageService)  # type: ignore
user_query_handler = ioc_container.resolve(UserQueryHandler)  # type: ignore


@router.post("/api/admin/user/list")
async def user_list(request: Request):
    """用户列表"""
    await request_auth_helper.require_login_user_admin(request)

    request_json = await request_value_helper.get_request_json(request)
    page_num = request_value_helper.get_request_json_int(request_json, 1, "pageNum")
    page_size = request_value_helper.get_request_json_int(request_json, 1, "pageSize")
    # criteria_json = request_json.get("criteria", {})

    criteria = UserQueryCriteria()
    # criteria.keyword = request_value_helper.get_request_json_string_trim(criteria_json, "", "keyword")
    criteria.keyword = request_value_helper.get_request_json_string_trim_or_empty(request_json, "criteria", "keyword")

    total_count = await user_query_handler.query_count(criteria)
    page_num, page_size, total_count = request_value_helper.fix_paging(page_num, page_size, total_count)
    list = await user_query_handler.query_by_page(page_num, page_size, criteria)
    return Result.ok(data={
        # "list": [user.to_json() for user in list],
        "list": list,
        "paging": {
            "pageNum": page_num,
            "pageSize": page_size,
            "totalCount": total_count
        }
    })


@router.get("/api/admin/user/get")
async def user_get(request: Request, id: str):
    """用户详情"""
    await request_auth_helper.require_login_user_admin(request)

    user_dto = await user_query_handler.query_by_id_req(id)
    # return Result.ok(data=user_dto.to_json())
    return Result.ok(data={"detail": user_dto})


@router.post("/api/admin/user/create")
async def user_create(request: Request):
    """创建用户"""
    await request_auth_helper.require_login_user_admin(request)

    request_json = await request_value_helper.get_request_json(request)
    username = request_value_helper.get_request_json_string_trim_req(request_json, "username")
    password = request_value_helper.get_request_json_string_trim_req(request_json, "password")
    nickname = request_value_helper.get_request_json_string_trim_req(request_json, "nickname")
    mobile = request_value_helper.get_request_json_string_trim_or_empty(request_json, "mobile")

    user_id = await user_manage_service.create_user(username, password, nickname, mobile)
    return Result.ok(data={"id": user_id})


@router.post("/api/admin/user/modify")
async def user_modify(request: Request):
    """修改用户"""
    await request_auth_helper.require_login_user_admin(request)

    request_json = await request_value_helper.get_request_json(request)
    id = request_value_helper.get_request_json_string_trim_req(request_json, "id")
    username = request_value_helper.get_request_json_string_trim_req(request_json, "username")
    nickname = request_value_helper.get_request_json_string_trim_req(request_json, "nickname")
    mobile = request_value_helper.get_request_json_string_trim_or_empty(request_json, "mobile")

    await user_manage_service.modify_user(id, username, nickname, mobile)
    return Result.ok()


@router.post("/api/admin/user/remove")
async def user_remove(request: Request):
    """删除用户"""
    await request_auth_helper.require_login_user_admin(request)

    request_json = await request_value_helper.get_request_json(request)
    id = request_value_helper.get_request_json_string_trim_req(request_json, "id")

    await user_manage_service.remove_user(id)
    return Result.ok()

File: dotnet/src/adapter/driving/UserManageController.cs
using DddExample.Application;

using Microsoft.AspNetCore.Mvc;

namespace DddExample.Adapter.Driving
{
    /// <summary>
    /// 用户管理 Controller
    /// </summary>
    public class UserManageController : ControllerBase
    {
        private readonly UserQueryHandler userQueryHandler;
        private readonly UserManageService userManageService;

        public UserManageController(UserQueryHandler userQueryHandler, UserManageService userManageService)
        {
            this.userQueryHandler = userQueryHandler;
            this.userManageService = userManageService;
        }

        /// <summary>
        /// 用户列表
        /// </summary>
        [HttpPost("/api/admin/user/list")]
        public async Task<Result> UserList()
        {
            await RequestAuthHelper.RequireLoginUserAdmin(Request);

            var requestJson = await RequestValueHelper.GetRequestJsonAsync(Request);
            int pageNum = RequestValueHelper.GetRequestJsonInt(requestJson, 1, "pageNum");
            int pageSize = RequestValueHelper.GetRequestJsonInt(requestJson, 1, "pageSize");

            //JsonNode? criteriaJson = requestJson["criteria"];
            UserQueryCriteria criteria = new UserQueryCriteria();
            //if (criteriaJson != null)
            //{
            //string? keyword = criteriaJson["keyword"]?.GetValue<string?>()?.Trim();
            string keyword = RequestValueHelper.GetRequestJsonStringTrimOrEmpty(requestJson, "criteria", "keyword");
            criteria.Keyword = keyword;
            //}

            int totalCount = await this.userQueryHandler.QueryCount(criteria);
            var paging = RequestValueHelper.FixPaging(pageNum, pageSize, totalCount);
            IList<UserDto> list = await this.userQueryHandler.QueryByPage(paging.pageNum, paging.pageSize, criteria);
            return Result.OkData(new
            {
                list = list,
                paging = new
                {
                    pageNum = paging.pageNum,
                    pageSize = paging.pageSize,
                    totalCount = paging.totalCount
                }
            });
        }

        /// <summary>
        /// 用户详情
        /// </summary>
        [HttpGet("/api/admin/user/get")]
        public async Task<Result> UserGet()
        {
            await RequestAuthHelper.RequireLoginUserAdmin(Request);

            string id = RequestValueHelper.GetRequestParamStringTrimReq(Request, "id");
            UserDto userDto = await this.userQueryHandler.QueryByIdReq(id);
            return Result.OkData(new
            {
                detail = userDto
            });
        }

        /// <summary>
        /// 创建用户
        /// </summary>
        [HttpPost("/api/admin/user/create")]
        public async Task<Result> UserCreate()
        {
            await RequestAuthHelper.RequireLoginUserAdmin(Request);

            var requestJson = await RequestValueHelper.GetRequestJsonAsync(Request);
            string username = RequestValueHelper.GetRequestJsonStringTrimReq(requestJson, "username");
            string password = RequestValueHelper.GetRequestJsonStringTrimReq(requestJson, "password");
            string nickname = RequestValueHelper.GetRequestJsonStringTrimReq(requestJson, "nickname");
            string mobile = RequestValueHelper.GetRequestJsonStringTrimOrEmpty(requestJson, "mobile");

            string userId = await this.userManageService.CreateUser(username, password, nickname, mobile);
            return Result.OkData(new
            {
                id = userId
            });
        }

        /// <summary>
        /// 修改用户
        /// </summary>
        [HttpPost("/api/admin/user/modify")]
        public async Task<Result> UserModify()
        {
            await RequestAuthHelper.RequireLoginUserAdmin(Request);

            var requestJson = await RequestValueHelper.GetRequestJsonAsync(Request);
            string id = RequestValueHelper.GetRequestJsonStringTrimReq(requestJson, "id");
            string username = RequestValueHelper.GetRequestJsonStringTrimReq(requestJson, "username");
            string nickname = RequestValueHelper.GetRequestJsonStringTrimReq(requestJson, "nickname");
            string mobile = RequestValueHelper.GetRequestJsonStringTrimOrEmpty(requestJson, "mobile");

            await this.userManageService.ModifyUser(id, username, nickname, mobile);
            return Result.Ok();
        }


        /// <summary>
        /// 删除用户
        /// </summary>
        [HttpPost("/api/admin/user/remove")]
        public async Task<Result> UserRemove()
        {
            await RequestAuthHelper.RequireLoginUserAdmin(Request);

            var requestJson = await RequestValueHelper.GetRequestJsonAsync(Request);
            string id = RequestValueHelper.GetRequestJsonStringTrimReq(requestJson, "id");

            await this.userManageService.RemoveUser(id);
            return Result.Ok();
        }
    }
}